### DISCUSSION TITLE
Unauthorized redirect location contains current path params?

### DISCUSSION THREAD

**Discussion by WilliamMayor** [maxcountryman/flask-login#658] [Q&A]
Hey,

I was told to cross post this here: https://github.com/maxcountryman/flask-login/issues/657

I did genuinely think it was a bug? Or at least unintended side effects?

Is this working as intended? Should I change our integration?

**Comment by 0xOliver** [comment]
Hello,

I have the same observation.
My local workaround, probably not a common solution but fixed my failing tests:

```diff
diff --git a/src/flask_login/utils.py b/src/flask_login/utils.py
index 44e7168..68656a0 100644
--- a/src/flask_login/utils.py
+++ b/src/flask_login/utils.py
@@ -94,7 +94,7 @@ def expand_login_view(login_view):
     if login_view.startswith(("https://", "http://", "/")):
         return login_view
     else:
-        if request.view_args is None:
+        if request.view_args is None or request.url_rule.subdomain == '':
             return url_for(login_view)
         else:
             return url_for(login_view, **request.view_args)
```

**Comment by alanhamlett** [comment]
Merged #663 to only add view args defined by the current route's `subdomain` or `host` variables.
